pkg/sentry/devices/nvproxy/frontend.go

```diff
@@ -587,6 +587,28 @@ func ctrlClientSystemGetBuildVersion(fi *frontendIoctlState, ioctlParams *nvgpu.
 	return n, nil
 }
 
+func ctrlDevGpuGetClasslist(fi *frontendIoctlState, ioctlParams *nvgpu.NVOS54Parameters) (uintptr, error) {
+	var ctrlParams nvgpu.NV0080_CTRL_CMD_GPU_GET_CLASSLIST_PARAMS
+
+	if _, err := ctrlParams.CopyIn(fi.t, addrFromP64(ioctlParams.Params)); err != nil {
```

Comment:
I'm missing something here, because if I assert on the correct param size as other command implementations do, I get an assertion failure as `ioctlParams.ParamsSize` is only `0x20`. I don't yet know why this is happening.

Reply:
It looks like the `request` argument is just incorrectly encoding the size of the `*argp` argument. I can make a weaker `ctrlParams.SizeBytes() >= int(ioctlParams.ParamsSize)` assertion and leave a note.

Reply:
With 525.105.17 and a `ctrlParams.SizeBytes() != int(ioctlParams.ParamsSize)` check here, your repro passes for me:

```
$ sudo docker run -e NVIDIA_VISIBLE_DEVICES=all --runtime=runsc-d -v ~:/hosthome -it repro-oom
2.1.1+cu121
matrix size: 50000 x 50000
Traceback (most recent call last):
  File "repro.py", line 8, in <module>
    N = torch.linalg.inv(M)
torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 18.63 GiB. GPU 0 has a total capacty of 21.99 GiB of which 3.18 GiB is free. Process 3116465 has 18.81 GiB memory in use. Of the allocated memory 18.63 GiB is allocated by PyTorch, and 0 bytes is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF
```

Note that if you're looking at debug log messages like:

```
I1115 21:33:45.646745  3116465 strace.go:567] [   1:   1] python3 E ioctl(0xa /dev/nvidiactl, 0xc020462a, 0x7ee0b25ab040)
D1115 21:33:45.646751  3116465 frontend.go:154] [   1:   1] nvproxy: frontend ioctl: nr = 0x0000002a, argSize = 0x00000020
D1115 21:33:45.646758  3116465 frontend.go:488] [   1:   1] nvproxy: control command 0x800201
I1115 21:33:45.646779  3116465 strace.go:605] [   1:   1] python3 X ioctl(0xa /dev/nvidiactl, 0xc020462a, 0x7ee0b25ab040) = 0 (0x0) (27.494µs)
```

argSize 0x20 (32 bytes) is the size of NVOS54_PARAMETERS, the parameter to NV_ESC_RM_CONTROL, not the size of the *control command*'s parameters. I think it's correct for NV0080_CTRL_CMD_GPU_GET_CLASSLIST_PARAMS to be 16 bytes.

Reply:
Ah thanks for investigating Jamie!

I guess we can now add back the size check.

Reply:
Oh yes thanks, I was making the mistake of thinking the `NVOS54_PARAMETERS` size was relevant here.